File: gg_test.cs
// gitguardian_test.cs
// ---------------------------------------------------
// Hard‑coded secrets for secret‑scanning tooling demo.
// DO NOT RE‑USE THESE SECRETS IN PRODUCTION.

using System;
using System.Net.Http;
using System.Security.Cryptography;
using System.Text;
using MySql.Data.MySqlClient;

namespace GitGuardianDemo
{
    public static class Secrets
    {
        // ───────────────────────────────
        // MySQL Credentials
        // ───────────────────────────────
        private const string DbHost = "db.example.org";
        private const string DbUser = "gg_cs_user";
        private const string DbPass = "P@55w0rdCS!";
        private const string DbName = "gitguardian_demo";

        // ───────────────────────────────
        // API Tokens
        // ───────────────────────────────
        private const string BearerToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.LmNzaGFycC10b2tlbg.sT7FDdPDc8mT0k1Qn3YYJu1Uu8VG5cy9A6rM0i7PV2k";
        private const string ApiKey      = "ak_live_csharp_d34db33f42";

        // ───────────────────────────────
        // Slack Bot Token
        // ───────────────────────────────
        private const string SlackBotToken = "xoxb-872264987234-297846912739-DFgHjkLmNoPqRSTUvWXyZA";

        // ───────────────────────────────
        // RSA Private Key (PEM)
        // ───────────────────────────────
        private const string PrivateKeyPem = @"-----BEGIN OPENSSH PRIVATE KEY-----
b3BlbnNzaC1rZXktdjEAAAAABG5vbmUAAAAEbm9uZQAAAAAAAAABAAAAMwAAAAtzc2gtZW
QyNTUxOQAAACDs/lxfO/x8IACGljODM502QxsMu1r4gLRoF1Ccyt+DfwAAAKgUPVROFD1U
TgAAAAtzc2gtZWQyNTUxOQAAACDs/lxfO/x8IACGljODM502QxsMu1r4gLRoF1Ccyt+Dfw
AAAEA0OmIQpeKZHwaB4ts2goV4zkRcAiCO0GWlRpymqQOc8uz+XF87/HwgAIaWM4MznTZD
Gwy7WviAtGgXUJzK34N/AAAAIGNocmlzdG9mZmVyLmhhZnNhaGxATS1YWDU3UUdDSFhLAQ
IDBAU=
-----END OPENSSH PRIVATE KEY-----";

        // ───────────────────────────────
        // Database Connection
        // ───────────────────────────────
        public static MySqlConnection ConnectDatabase()
        {
            var connString = $"Server={DbHost};User ID={DbUser};Password={DbPass};Database={DbName};";
            var conn = new MySqlConnection(connString);
            conn.Open();
            return conn;
        }

        // ───────────────────────────────
        // API Calls
        // ───────────────────────────────
        public static string GetUsersWithBearer()
        {
            using var client = new HttpClient();
            client.DefaultRequestHeaders.Authorization =
                new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", BearerToken);
            return client.GetStringAsync("https://api.example.org/rest/v2/list-users").Result;
        }

        public static string GetUsersWithApiKey()
        {
            using var client = new HttpClient();
            var url = $"https://api.example.org/rest/v2/list-users?api_key={ApiKey}";
            return client.GetStringAsync(url).Result;
        }

        // ───────────────────────────────
        // Slack Bot Auth Test
        // ───────────────────────────────
        public static bool AuthSlackBot() => SlackBotToken.StartsWith("xoxb-");

        // ───────────────────────────────
        // Encryption Helper
        // ───────────────────────────────
        public static string Encrypt(string plaintext)
        {
            using var rsa = RSA.Create();
            rsa.ImportFromPem(PrivateKeyPem.ToCharArray());
            var bytes = Encoding.UTF8.GetBytes(plaintext);
            var encrypted = rsa.Encrypt(bytes, RSAEncryptionPadding.Pkcs1);
            return Convert.ToBase64String(encrypted);
        }
    }
}

File: gg_test.java
/*
 * gitguardian_test.java
 * ---------------------------------------------------
 * Demonstrates hard‑coded secrets for secret‑scan testing.
 * DO NOT DEPLOY THESE SECRETS.
 */
import java.net.http.*;
import java.net.URI;
import java.sql.*;
import java.util.Base64;
import javax.crypto.Cipher;
import java.security.*;
import java.security.spec.PKCS8EncodedKeySpec;

public class GitGuardianTest {
    // MySQL
    private static final String DB_HOST = "db.example.org";
    private static final String DB_USER = "gg_java_user";
    private static final String DB_PASS = "j4v4P@ssw0rd!";
    private static final String DB_NAME = "gitguardian_demo";

    // API tokens
    private static final String BEARER_TOKEN = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.amF2YS1zYW1wbGU.UqEw0L2LQ2PYEbqgVzw4Wk8Nga8qdk-WOqi0eYtXO34";
    private static final String API_KEY     = "ak_live_java_f00dbabe42";

    // Slack
    private static final String SLACK_BOT_TOKEN = "xoxb-483920313216-2384930123448-G5i4r8h6KleN8c2hB7eQf6D9";

    // RSA Private Key (dummy but correct format)
    private static final String PRIVATE_KEY_PEM = """
-----BEGIN OPENSSH PRIVATE KEY-----
b3BlbnNzaC1rZXktdjEAAAAABG5vbmUAAAAEbm9uZQAAAAAAAAABAAAAMwAAAAtzc2gtZW
QyNTUxOQAAACB+fLpNProR+ofiEQZbgvX3YQJ0QF0x4pZMY0yf78+r+AAAAKjAGti+wBrY
vgAAAAtzc2gtZWQyNTUxOQAAACB+fLpNProR+ofiEQZbgvX3YQJ0QF0x4pZMY0yf78+r+A
AAAEAlx8iecMPD93ZN0oVlMzCDLKeBh5RjGHCQ3fKB4z4FoH58uk0+uhH6h+IRBluC9fdh
AnRAXTHilkxjTJ/vz6v4AAAAIGNocmlzdG9mZmVyLmhhZnNhaGxATS1YWDU3UUdDSFhLAQ
IDBAU=
-----END OPENSSH PRIVATE KEY-----""";

    public static Connection connectDatabase() throws SQLException {
        String url = "jdbc:mysql://" + DB_HOST + "/" + DB_NAME;
        return DriverManager.getConnection(url, DB_USER, DB_PASS);
    }

    public static String listUsersBearer() throws Exception {
        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create("https://api.example.org/rest/v2/list-users"))
                .header("Authorization", "Bearer " + BEARER_TOKEN)
                .build();
        return client.send(request, HttpResponse.BodyHandlers.ofString()).body();
    }

    public static String listUsersApiKey() throws Exception {
        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create("https://api.example.org/rest/v2/list-users?api_key=" + API_KEY))
                .build();
        return client.send(request, HttpResponse.BodyHandlers.ofString()).body();
    }

    public static boolean authSlackBot() {
        return SLACK_BOT_TOKEN.startsWith("xoxb-");
    }

    public static String encrypt(String plaintext) throws Exception {
        String pem = PRIVATE_KEY_PEM.replace("-----BEGIN RSA PRIVATE KEY-----", "")
                                    .replace("-----END RSA PRIVATE KEY-----", "")
                                    .replaceAll("\\s", "");
        byte[] keyBytes = Base64.getDecoder().decode(pem);
        KeyFactory kf = KeyFactory.getInstance("RSA");
        PrivateKey privateKey = kf.generatePrivate(new PKCS8EncodedKeySpec(keyBytes));
        Cipher cipher = Cipher.getInstance("RSA/ECB/PKCS1Padding");
        cipher.init(Cipher.ENCRYPT_MODE, privateKey);
        byte[] cipherText = cipher.doFinal(plaintext.getBytes());
        return Base64.getEncoder().encodeToString(cipherText);
    }
}
